haystack/document_stores/types/filter_policy.py

```diff
@@ -81,6 +81,19 @@ def convert_logical_operators(filter_dict: Dict[str, Any]) -> Dict[str, Any]:
     return filter_dict
 
 
+def logical_operators_to_str(filter_dict: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Convert any logical operators found in a filter dictionary to string.
+
+    :param filter_dict: A dictionary representing a filter with potential LogicalOperator enums.
+    :return: A new dictionary with LogicalOperator strings in place of enums.
+    """
+    # If the dictionary represents a logical filter, update the 'operator'
+    if is_logical_filter(filter_dict) and isinstance(filter_dict["operator"], LogicalOperator):
+        filter_dict["operator"] = filter_dict["operator"].value
+    return filter_dict
```

Comment:
Why are we introducing a new function here instead of just calling `str(logical_operator)` when adding it to the combined dict? 

Correct me if I'm wrong, but the only time the enum ever comes into play is inside the `apply_filters` function, i.e., we can assume that `init_filters` and `runtime_filters` will only contain strings in the `operator` field/key.

Reply:
We can do it that way as well. The current approach seemed less demanding for maintenance but no problem, I'll make adjustments.

Reply:
Removed function. [aeac10d](https://github.com/deepset-ai/haystack/pull/8042/commits/aeac10dc4b26d1f0410b178f1a8ae8cdbaef2b41) ensures that every path of execution that might return logical filter returns it with the str operator representation.